controllers/account.js

```diff
@@ -30,6 +39,30 @@ module.exports.renderChangeProfile = async (req, res) => {
   });
 };
 
+module.exports.updatePassword = async (req, res) => {
+  User.findById(req.user._id, (err, user) => {
+    if (err) {
+      res.send(err);
+    } else {
+      if (req.body.newPassword === req.body.repeatPassword) {
+        user.changePassword(
+          req.body.oldPassword,
+          req.body.newPassword,
+          (err) => {
+            if (err) {
+              //error message
+              res.redirect("/account/change-password");
+            } else {
+              //password changed successfully message
+              res.redirect("/account");
+            }
```

Comment:
Seems like the app hangs if you leave one of the fields empty for me. Should be some kind of error handling for these cases as well.